cura/Settings/ContainerManager.py

```diff
@@ -712,14 +712,21 @@ def createMaterial(self) -> str:
         if not global_stack:
             return ""
 
-        containers = self._container_registry.findInstanceContainers(id="generic_pla")
+        approximate_diameter = round(global_stack.getProperty("material_diameter","value"))
+        containers = self._container_registry.findInstanceContainers(id="generic_pla*", approximate_diameter=approximate_diameter)
```

Comment:
A few code style problems here. I fixed them for you in f65043ecc8f7b506464b31ba5995fac58b4d0358.